clustering/k-means-clustering-plugin/tests/test_main.py

```diff
@@ -0,0 +1,247 @@
+        self.dirpath = pathlib.Path.cwd()
+        self.inp_dir = tempfile.mkdtemp(dir=self.dirpath)
+        self.out_dir = tempfile.mkdtemp(dir=self.dirpath)
+        self.file_pattern = file_pattern
+        self.size = size
+        self.outname = outname
+        self.x = self.create_dataframe()
+
+    def get_inp_dir(self) -> pathlib.Path:
+        """Get input directory."""
+        return pathlib.Path(self.inp_dir)
+
+    def get_out_dir(self) -> pathlib.Path:
+        """Get output directory."""
+        return pathlib.Path(self.out_dir)
+
+    def create_dataframe(self) -> pd.core.frame.DataFrame:
+        """Create Pandas dataframe."""
+        diction_1 = {
+            "A": np.linspace(0.0, 4.0, self.size, dtype="float32", endpoint=False),
+            "B": np.linspace(0.0, 6.0, self.size, dtype="float32", endpoint=False),
+            "C": np.linspace(0.0, 8.0, self.size, dtype="float32", endpoint=False),
+            "D": np.linspace(0.0, 10.0, self.size, dtype="float32", endpoint=False),
+            "label": np.random.randint(low=1, high=4, size=self.size),
+        }
+
+        df = pd.DataFrame(diction_1)
+
+        return df
+
+    def csv_func(self) -> None:
+        """Convert pandas dataframe to csv file format."""
+        self.x.to_csv(pathlib.Path(self.inp_dir, self.outname), index=False)
+
+    def parquet_func(self) -> None:
+        """Convert pandas dataframe to parquet file format."""
+        self.x.to_parquet(
+            pathlib.Path(self.inp_dir, self.outname), engine="auto", compression=None
+        )
+
+    def feather_func(self) -> None:
+        """Convert pandas dataframe to feather file format."""
+        self.x.to_feather(pathlib.Path(self.inp_dir, self.outname))
+
+    def arrow_func(self) -> None:
+        """Convert pandas dataframe to Arrow file format."""
+        self.x.to_feather(pathlib.Path(self.inp_dir, self.outname))
+
+    def hdf_func(self) -> None:
+        """Convert pandas dataframe to hdf5 file format."""
+        v_df = vaex.from_pandas(self.x, copy_index=False)
+        v_df.export(pathlib.Path(self.inp_dir, self.outname))
+
+    def __call__(self) -> None:
+        """To make a class callable."""
+        data_ext = {
+            ".hdf5": self.hdf_func,
+            ".csv": self.csv_func,
+            ".parquet": self.parquet_func,
+            ".feather": self.feather_func,
+            ".arrow": self.arrow_func,
+        }
+
+        return data_ext[self.file_pattern]()
+
+    def clean_directories(self):
+        """Remove files."""
+        for d in self.dirpath.iterdir():
+            if d.is_dir() and d.name.startswith("tmp"):
+                shutil.rmtree(d)
+
+
+@pytest.fixture(
+    params=[
+        (".csv", 2, 5),
+        (".arrow", 2, 7),
+        (".feather", 2, 20),
+        (".hdf5", 2, 30),
+        (".parquet", 2, 10),
+    ]
+)
+def get_params(request):
+    """To get the parameter of the fixture."""
+    yield request.param
```

Comment:
I think this can be simplified by just parametrizing `test_elbow`. If you're just generating a set of inputs like this, you can just use `pytest.mark.parametrize`. That way you don't need to generate an intermediate fixture.

Fixtures are generally more useful when you want to use the same set of inputs for multiple tests. For example, a fixture might be a list of all k-selection methods. Another fixture might be a list of minrange and maxrange pairs. Then, when you use the k-selection fixture and range the fixture together, it will iterate over all possible combinations of those two fixtures.